plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitManager.java

```diff
@@ -68,7 +98,16 @@ public ConnectorSplitSource getSplits(
                                 // (See AbstractTestIcebergSmoke#testLargeInFailureOnPartitionedColumns)
                                 .intersect(table.getUnenforcedPredicate().simplify(ICEBERG_DOMAIN_COMPACTION_THRESHOLD))))
                 .useSnapshot(table.getSnapshotId().get());
-        IcebergSplitSource splitSource = new IcebergSplitSource(table.getSchemaTableName(), tableScan.planTasks());
+        IcebergSplitSource splitSource = new IcebergSplitSource(
+                table.getSchemaTableName(),
+                identityPartitionColumns,
+                tableScan.planTasks(),
+                dynamicFilter,
+                session.getTimeZoneKey(),
+                maxSplitsPerSecond,
+                maxOutstandingSplits,
+                new BoundedExecutor(executorService, splitLoaderConcurrency),
```

Comment:
I'm not sure this needs to be configurable, might just want to hardcode the thread count.

Reply:
Can we hard-code splitLoaderConcurrency to 1 given that we're not parallelising split loading in IcebergSplitSource ?

Reply:
That, or we could just pass the ExecutorService straight through without wrapping in BoundedExecutor

Reply:
It's not completely single threaded because the AsyncQueue uses the executor to do some of its operations.

Reply:
Wouldn't setting `splitLoaderConcurrency` to 1 result in a deadlock, as we would use one thread for:
```
addExceptionCallback(ResumableTasks.submit(executor, loaderTask), this::onLoaderFailure);
```

And then there would be no threads left for `splitQueue` handling:
```
new ThrottledAsyncQueue<>(maxSplitsPerSecond, maxOutstandingSplits, executor);
```